crates/core/tedge_api/src/workflow/error.rs

```diff
@@ -33,6 +36,16 @@ pub enum ScriptDefinitionError {
     IncorrectRange { from: u8, to: u8 },
 }
 
+/// Error related to state excerpt definitions
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum StateExcerptError {
+    #[error("Only records can be used for command input/output, not a {kind} as {value}")]
+    NotAnObject {
+        kind: String,
+        value: serde_json::Value,
+    },
+}
```

Comment:
note: IMO instead of putting all error types into `error` module, it would be better to place them together with the thing that can potentially return them, so in this case, `StateExcerpt`